CHANGELOG.md

```diff
@@ -6,7 +6,147 @@ document.
+* [`len_zero`] now also triggers if deref target implements `is_empty()`
+  [#13871](https://github.com/rust-lang/rust-clippy/pull/13871)
+* [`arbitrary_source_item_ordering`] now makes alphabetic ordering in module item groups optional (off by default)
+  [#13718](https://github.com/rust-lang/rust-clippy/pull/13718)
+* [`ptr_eq`] now handles more cases, including `!=` in addition to `==`
+  [#14339](https://github.com/rust-lang/rust-clippy/pull/14339)
+* [`struct_field_names`] now also checks private fields of public structs
+  [#14076](https://github.com/rust-lang/rust-clippy/pull/14076)
+* [`never_loop`] applicability is now `MachineApplicable` when the loop block contains neither `break` nor `continue`
+  [#14203](https://github.com/rust-lang/rust-clippy/pull/14203)
+* [`if_then_some_else_none`] now has `MachineApplicable` applicability
+  [#14106](https://github.com/rust-lang/rust-clippy/pull/14106)
+* Improve `string_to_string` lint in map calls
+  [#14396](https://github.com/rust-lang/rust-clippy/pull/14396)
+* [`single_match`], [`single_match_else`] reinstated for matches with comments (non-autofix)
+  [#14420](https://github.com/rust-lang/rust-clippy/pull/14420)
+* [`manual_strip`] now reuses existing identifier in suggestion when possible
+  [#14188](https://github.com/rust-lang/rust-clippy/pull/14188)
+* [`needless_pass_by_value`] suggests using a reference on the innermost `Option` content
+  [#14392](https://github.com/rust-lang/rust-clippy/pull/14392)
+* [`needless_pass_by_value`] improves suggestions for `Option` types
+  [#13880](https://github.com/rust-lang/rust-clippy/pull/13880)
+* [`obfuscated_if_else`] now supports `then().unwrap_or_else()` and `then_some().unwrap_or_else()`
+  [#14165](https://github.com/rust-lang/rust-clippy/pull/14165)
+* [`mixed_case_hex_literals`] now adds alternative suggestions
+  [#14235](https://github.com/rust-lang/rust-clippy/pull/14235)
+* [`range_zip_with_len`] now has an autofix
+  [#14136](https://github.com/rust-lang/rust-clippy/pull/14136)
+* Format macros: all format-handling lints now validate `todo!` and `unimplemented!` macros
+  [#14266](https://github.com/rust-lang/rust-clippy/pull/14266)
+* [`disallowed_methods`] now supports replacements
+  [#13669](https://github.com/rust-lang/rust-clippy/pull/13669)
+* [`io_other_error`] Fix non-applicable suggestion
+  [#14349](https://github.com/rust-lang/rust-clippy/pull/14349)
+* Added MSRV checks for several lints:
+  * [`question_mark`] [#14436](https://github.com/rust-lang/rust-clippy/pull/14436)
+  * [`repeat_vec_with_capacity`] [#14126](https://github.com/rust-lang/rust-clippy/pull/14126)
+  * [`manual_flatten`] [#14086](https://github.com/rust-lang/rust-clippy/pull/14086)
+  * [`lines_filter_map_ok`] [#14130](https://github.com/rust-lang/rust-clippy/pull/14130)
+  * [`incompatible_msrv`] now lints function calls with any argument count
+    [#14216](https://github.com/rust-lang/rust-clippy/pull/14216)
```

Comment:
Is this intentionally under "Added MSRV checks"? It seems like an improvement to a lint and maybe would be better as its own point under "Enhancements"